payment/order.go

```diff
@@ -113,37 +117,121 @@ func CreateOrderItemFromMacaroon(sku string, quantity int) (*OrderItem, error) {
 		caveat := mac.Caveats()[i]
 		values := strings.Split(string(caveat.Id), "=")
 		key := strings.TrimSpace(values[0])
-		value := strings.TrimSpace(values[1])
+		value := strings.TrimSpace(strings.Join(values[1:], "="))
```

Comment:
This seems confusing, I might be mistaken, but this joins a bunch of strings together on the first instance of a key?

Reply:
the caveats look like this after we pull them from the macaroon:

` key=value`

line 118 we are splitting on "=", then taking the first value as a key on 119.  On 120 we are reconstructing the right side by joining with the seperator of "=".

This will handle caveat values which include `=` in the value like the below:

`key=value=or=something=else`

where the key should be `key` and the value should be `value=or=something=else`